harness/determined/common/experimental/workspace.py

```diff
@@ -77,6 +79,22 @@ def get_with_offset(offset: int) -> bindings.v1ListRPsBoundToWorkspaceResponse:
 
         return resource_pools
 
+    def list_projects(self) -> List[project.Project]:
+        """Lists the projects that are a part of this workspace."""
+
+        def get_with_offset(offset: int) -> bindings.v1GetWorkspaceProjectsResponse:
+            return bindings.get_GetWorkspaceProjects(
+                session=self._session,
+                offset=offset,
+                id=self.id,
+            )
+
+        bindings_projects: Iterable[bindings.v1Project] = itertools.chain.from_iterable(
```

Comment:
[Style guide](https://google.github.io/styleguide/pyguide.html#274-decision) says:
> Multiple for clauses or filter expressions are not permitted. Use loops instead when things get more complicated.

I agree with it here. Nested "for"s are always hard for me to read.

I chose the `itertools` route here over a for loop because the object this line creates is really natural. Honestly, I think something like this is what `read_paginated` should return in the first place.

Which, hmm... I don't think that'd be hard to write. `limit` and `offset` would need some accommodation, but doing this still would be better.

Reply:
i agree nested for loops are unideal. but this basically splits up the nesting into two steps, which could also be done with list comprehension?
`bindings_projects = [r.projects for r in api.read_paginated(get_with_offset)]`?

i guess i tend to prefer minimal imports/dependencies (though itertools seems harmless). also `itertools` seems less readable than list comprehension. i have no idea if that does something special without looking it up.

either way, not a big deal.

Reply:
If you like, sure. Done.

Reply:
Well... undone. The type of `[r.projects for r in api.read_paginated(get_with_offset)]` is a `[[bindings.v1Project]]`, not a `[bindings.v1Project]` (list of lists, not a list). Flattening the nesting is one of the nice things `chain.from_iterable` does.

If you're less a fan of `itertools.chain.from_iterable`, then I can replace it with an accumulator:
```python
projects  = []
for binding_projects in api.read_paginated(get_with_offset):
  for p in binding_projects:
    projects.append(project.Project._from_bindings(p, self._session)
```

Reply:
nah, makes sense.